### PR TITLE
Select Team Pokemon

### PR DESCRIPTION
**Pull request by naftalima**
## Objective and Motivation
- The trainer selects the 3 pokemons for her/his team.

### CODE REVIEW

File: backend/battles/views.py [hunk 1]
```diff
@@ -18,23 +19,26 @@ class CreateBattleView(CreateView):
...
+        # BUG can't challenged yourself
+        team_creator = Team.objects.create(battle=battle, trainer=form.instance.creator)
         Team.objects.create(battle=battle, trainer=form.instance.opponent)
 
-        response = super(CreateBattleView, self).form_valid(form)
-        return response
+        return HttpResponseRedirect(reverse_lazy("battle-team-pokemons", args=(team_creator.id,)))
 
 
 # HACK : should be a UpdateView
```
Comment (rsarai) [inline backend/battles/views.py:36]: You can already remove this comment

File: backend/battles/views.py [hunk 2]
```diff
@@ -18,23 +19,26 @@ class CreateBattleView(CreateView):
...
-class SelectTeamView(TemplateView):
+class SelectTeamView(UpdateView):
+    model = Team
     template_name = "battles/battle-team-pokemons.html"
+    form_class = TeamForm
+    success_url = reverse_lazy("battles")
 
     # TODO : do the validation on the form
```
Comment (rsarai) [inline backend/battles/views.py:43]: Are these comments necessary?

File: backend/battles/views.py [hunk 3]
```diff
@@ -18,23 +19,26 @@ class CreateBattleView(CreateView):
     model = Battle
     template_name = "battles/battle-opponent.html"
     form_class = BattleForm
-    success_url = reverse_lazy("battle-team-pokemons")
+    # success_url = reverse_lazy("battle-team-pokemons")
```
Comment (rsarai) [inline backend/battles/views.py:22]: You can remove this if is not being used

File: backend/battles/views.py [hunk 4]
```diff
@@ -18,23 +19,26 @@ class CreateBattleView(CreateView):
...
 
     def form_valid(self, form):
         # TODO init in form
         form.instance.creator = self.request.user
         battle = form.save()
 
-        Team.objects.create(battle=battle, trainer=form.instance.creator)
+        # BUG can't challenged yourself
```
Comment (rsarai) [inline backend/battles/views.py:29]: This comment can also be removed right?

File: backend/battles/views.py [hunk 5]
```diff
@@ -1,8 +1,9 @@
 from django.db.models import Q
+from django.http.response import HttpResponseRedirect
 from django.urls import reverse_lazy
-from django.views.generic import CreateView, DetailView, ListView, TemplateView
+from django.views.generic import CreateView, DetailView, ListView, TemplateView, UpdateView
 
-from .forms import BattleForm
+from .forms import BattleForm, TeamForm
```
Comment (rsarai) [inline backend/battles/views.py:6]: Let's change this to use absolute imports

File: backend/battles/templatetags/battles_tags.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
+# custom_tags.py
+from django import template
+
+from ..models import Battle, Team
```
Comment (rsarai) [inline backend/battles/templatetags/battles_tags.py:4]: Let's change this to use absolute imports :pray:

File: backend/battles/templates/battles/battles.html [hunk 7]
```diff
@@ -25,9 +23,10 @@ <h2 class="title">On going Battles</h2>
     {% endfor %}
   </table>
 
+  {% comment %}
```
Comment (rsarai) [inline backend/battles/templates/battles/battles.html:26]: The following block of comments is required or can we remove it?

File: backend/battles/pokemon.py [hunk 8]
```diff
@@ -44,12 +44,11 @@ def get_pokemon(poke_id):
 def sum_points(pokemons):
     points = 0
     for pokemon in pokemons:
-        points += pokemon["attack"] + pokemon["defense"] + pokemon["hp"]
+        points += pokemon.attack + pokemon.defense + pokemon.hp
+    print(points)
```
Comment (rsarai) [inline backend/battles/pokemon.py:48]: :eyes:

File: backend/battles/forms.py [hunk 9]
```diff
@@ -1,9 +1,64 @@
-from django import forms
+from django.forms import ModelChoiceField, ModelForm, ValidationError
 
-from .models import Battle
+from pokemons.models import Pokemon
 
+from .models import Battle, Team, TeamPokemon
```
Comment (rsarai) [inline backend/battles/forms.py:5]: Let's change this to use absolute imports :pray:

**Review by rsarai — CHANGES_REQUESTED** [review]

File: backend/battles/pokemon.py (hunk 8 above)
Comment (naftalima) [reply backend/battles/pokemon.py]: IDK why but it only works that way  👁️👁️

File: backend/battles/services/logic_team_pokemon.py [hunk 10]
```diff
@@ -0,0 +1,10 @@
+def sum_points(pokemons):
```
Comment (rsarai) [inline backend/battles/services/logic_team_pokemon.py:1]: Did you discuss this with Flavio?

Comment (rsarai) [reply backend/battles/services/logic_team_pokemon.py:1]: I mean the name of the module

File: backend/battles/templatetags/battles_tags.py [hunk 11]
```diff
@@ -0,0 +1,18 @@
+# custom_tags.py
```
[the remark this answers is not in the document]
Comment (naftalima) [reply backend/battles/templatetags/battles_tags.py:1]: no 😅

File: backend/battles/services/logic_team_pokemon.py (hunk 10 above)
Comment (naftalima) [reply backend/battles/services/logic_team_pokemon.py:1]: i don't remember, but definitely can be improved

Comment (naftalima) [reply backend/battles/services/logic_team_pokemon.py:1]: yes 🥰

**Comment by naftalima** [comment]
I got confused when I solved the previous conflict 😕

File: backend/battles/templatetags/battles_tags.py [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+
+
+@register.simple_tag()
+def total_battles():
+    return Battle.objects.count()
+
+
+@register.filter()
+def get_team(battle, user):
+    team_id = Team.objects.get(battle=battle, trainer=user).id
```
Comment (fjsj) [inline backend/battles/templatetags/battles_tags.py:16]: This will generate new queries. Consider not making new queries on filter tags.

**Review by fjsj — APPROVED** [review]
LGTM, there are additional queries that can be optimized, but you can tackle this later.

**Pull request merged**